server/stycoon/reports.go

```diff
@@ -59,8 +64,11 @@ func (game *Game) fillProfiling(tick *int64) error {
 			rows.Scan(&profiling.Movement, &profiling.Attacks, &profiling.Trades, &profiling.Recipes, &profiling.Prices, &profiling.Constructions, &profiling.Report, &profiling.Total, &profiling.Overall, &profiling.At)
 			profiling.Tick = *tick
 		}
-		game.Reports.Profiling = append(game.Reports.Profiling, profiling)
+		newReports = append(newReports, profiling)
 	}
+	game.Ready.Lock()
+	defer game.Ready.Unlock()
```

Comment:
I would just unlock it directly after the call without the defer - in case if someone added some logic after the append, it does not need to be locked (the same for the other fields)